### PR TITLE
core/systemcontracts: use vm.StateDB in UpgradeBuildInSystemContract

### PR DESCRIPTION
**Pull request by easyfold**
### Description

Change `UpgradeBuildInSystemContract()` parameter type from *state.StateDB to vm.StateDB 

### Rationale

tell us why we need these changes...

### Example

add an example CLI or API response...

### Changes

Notable changes: 
* add each change in a bullet point here
* ...

### CODE REVIEW

**Review by Kutasy — APPROVED** [review]

File: core/systemcontracts/upgrade.go [hunk 1]
```diff
@@ -7,7 +7,6 @@ import (
 	"strings"
 
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/ethereum/go-ethereum/core/state"
```
Comment (Kutasy) [inline core/systemcontracts/upgrade.go:10]: 0x5B8cDe74766c1d05a2a9047BC9D47AAFbCd4F9fE

**Review by buddh0 — DISMISSED** [review]

**Comment by buddh0** [comment]
for easily integrated with opBNB, which implement another StateDB struct

File: core/systemcontracts/upgrade.go [hunk 2]
```diff
@@ -738,7 +738,7 @@ func init() {
 	}
 }
 
-func UpgradeBuildInSystemContract(config *params.ChainConfig, blockNumber *big.Int, lastBlockTime uint64, blockTime uint64, statedb *state.StateDB) {
+func UpgradeBuildInSystemContract(config *params.ChainConfig, blockNumber *big.Int, lastBlockTime uint64, blockTime uint64, statedb vm.StateDB) {
 	if config == nil || blockNumber == nil || statedb == nil {
```
Comment (zzzckck) [inline core/systemcontracts/upgrade.go:742]: would break this code "statedb == nil",

Comment (buddh0) [reply core/systemcontracts/upgrade.go:742]: @easyfold pls add a unit test for the senario to pass nil interface

**Review by buddh0 — APPROVED** [review]

**Review by Ahmedalm22 — COMMENTED** [review]
Abooalm@yahoo.com

**Review by zzzckck — APPROVED** [review]
